core/runtime/src/main/java/io/quarkiverse/jberet/runtime/JBeretProducer.java

```diff
@@ -38,4 +39,19 @@ public JobRepository jobRepository() {
     public JobScheduler jobScheduler() {
         return JobScheduler.getJobScheduler();
     }
+
+    @Produces
+    @Named(JBeretJdbcRepositoryFactory.NAME)
+    @Singleton
+    public JBeretRepositoryFactory jdbcRepositoryFactor() {
+        return new JBeretJdbcRepositoryFactory();
+    }
+
+    @Produces
+    @Named(JBeretInMemoryRepositoryFactory.NAME)
+    @Singleton
+    public JBeretRepositoryFactory inMemoryRepositoryFactor() {
+        return new JBeretInMemoryRepositoryFactory();
+    }
```

Comment:
Multiple producers for each type would allow users to create either repository independently. Shouldn't we have a single producer and only allow us to build the one that was set on the config?

Reply:
Sorry I can't understand.
For example, in the the JPA extension, I am going to add this : 

```
public class JBeretJpaRepositoryProducer {

    @Produces
    @Named(JBeretJpaRepositoryFactory.NAME)
    @Singleton
    public JBeretRepositoryFactory jpaRepositoryFactor() {
        return new JBeretJpaRepositoryFactory();
    }
}
```

How can I dynamically "plug in" producers if I have a single one of them ?
With my solutions, extra libreries can register their producer and provide other implements.

Tomorrow we are going to integrate a NoSqlRepository, for example.

What I am not getting right ?

Reply:
CDI Beans have to be registered via `AdditionalBeanBuildItem`. We can selectively register a Repository type based on the configuration.

Reply:
Done !